celery/canvas.py

```diff
@@ -227,8 +227,7 @@ def _merge(self, args=None, kwargs=None, options=None, force=False):
         options = options if options else {}
         if self.immutable and not force:
             return (self.args, self.kwargs,
-                    dict(self.options,
-                         **options) if options else self.options)
+                    dict(options, **self.options) if options else self.options)
```

Comment:
I should point this bit out specifically. It was previously expected that any options passed when freezing a task would override ones already set in the task object being frozen. This change flips that logic such that options already set on the signature will be used rather than any of the ones passed in. This change is critical for making sure group IDs flow down to chained tasks, but I'm a bit concerned about such a radical change. None of the tests seem to break AFAICT, but it would be good to get some thought from the maintainers here.

Reply:
does this break existing codes?

Reply:
I can't confidently say that it doesn't, but I'd need to spend a bit of time with the failing tests to be sure that this is (or isn't :crossed_fingers: ) the cause.

Reply:
AFAICT this doesn't break anything. I could potentially change it to look more like this if you'd like:

```python
new_options = dict(self.options, **{k: v for k, v in options.items() if k not in {"group"}})
```

That would ensure that we only retain specific options from the original task which we knew need to be retained across a freeze. Thoughts, @auvipy ?

Reply:
sounds good to me

Reply:
I'm pretty sure this also needs to be done for the mutable case on the lines below. Even if the signature is mutable, allowing the group_id to be overridden would break completion so we can't allow it to be changed. That's actually kind of nice since it means I can make a `new_options` regardless of the signature's mutability and build it like I laid out in my previous comment. I'll have some changes and unit tests for this to push soon.

Reply:
This is done